async_lua_ev.c

```diff
@@ -50,6 +57,69 @@ static int async_new(lua_State* L) {
     return 1;
 }
 
+struct async_serialized {
+  ev_async*       async;
+  struct ev_loop* loop;
+};
+
+static int async_sz_send(lua_State *L) {
+    void *udata = lua_touserdata(L, lua_upvalueindex(1));
+    struct async_serialized* data = (struct async_serialized*)udata;
+
+    ev_async_send(data->loop, data->async);
+
+    return 0;
+}
+
+/**
+ * Export serialized send for multi-threading signalization (low-level API).
+ *
+ * Return string.
+ *
+ * Usage:
+ *     string = Async.export(async, loop)
+ *
+ * [+1, -0, ?]
+ */
+static int async_export(lua_State *L) {
+    ev_async*       async  = check_async(L, 1);
+    struct ev_loop* loop   = *check_loop_and_init(L, 2);
+
+    struct async_serialized data;
+    data.async = async;
+    data.loop  = loop;
+    lua_pushlstring(L, (const char*)&data, sizeof(struct async_serialized));
+
+    return 1;
+}
+
+/**
+ * Import serialized send for multi-threading signalization (low-level API).
+ *
+ * Return a function which will act as async:send(loop) when called.
+ *
+ * Calling the function beyond the lifespan of either the exported loop or
+ * async object will result in undefined behavior (crash/corruption). The same
+ * applies with an ill-formed string.
+ *
+ * Usage:
+ *     async_send = Async.import(string)
+ *
+ * [+1, -0, ?]
+ */
+static int async_import(lua_State *L) {
+    size_t len;
+    const char* str = luaL_checklstring(L, 1, &len);
+    if ( len != sizeof(struct async_serialized) )
+      luaL_error(L, "invalid data size");
+
+    void* data = lua_newuserdata(L, sizeof(struct async_serialized));
+    memcpy(data, str, sizeof(struct async_serialized));
```

Comment:
How can we ensure that the memory pointed to by the fields of the async_serialized structure are still valid?